tests/test_config_generator.py

```diff
@@ -9,24 +8,22 @@
 
 from patroni.__main__ import main as _main
 from patroni.config import Config
-from patroni.config_generator import AbstractConfigGenerator, get_address
-
+from patroni.config_generator import AbstractConfigGenerator, get_address, NO_VALUE_MSG
 from patroni.utils import patch_config
 
 from . import psycopg_connect
 
 
 @patch('patroni.psycopg.connect', psycopg_connect)
-@patch('socket.getaddrinfo', Mock(return_value=[(0, 0, 0, 0, ('1.9.8.4', 1984))]))
 @patch('builtins.open', MagicMock())
 @patch('subprocess.check_output', Mock(return_value=b"postgres (PostgreSQL) 16.2"))
 @patch('psutil.Process.exe', Mock(return_value='/bin/dir/from/running/postgres'))
 @patch('psutil.Process.__init__', Mock(return_value=None))
 class TestGenerateConfig(unittest.TestCase):
-
-    no_value_msg = '#FIXME'
-    _HOSTNAME = socket.gethostname()
-    _IP = sorted(socket.getaddrinfo(_HOSTNAME, 0, socket.AF_UNSPEC, socket.SOCK_STREAM, 0), key=lambda x: x[0])[0][4][0]
+    _HOSTNAME = 'tests_hostname'
+    _IP = '1.9.8.4'
+    setattr(AbstractConfigGenerator, '_HOSTNAME', _HOSTNAME)
+    setattr(AbstractConfigGenerator, '_IP', _IP)
```

Comment:
Why not use patch.object() for that?
Of course it would require to make _HOSTNAME and _IP global.

Reply:
right, no reason not to do that:)